### PR TITLE
Support installing Ruby 2.1.5 on Solaris 11

### PR DESCRIPTION
**Pull request by schisamo**
/cc @curiositycasualty @scotthain

### CODE REVIEW

**Comment by scotthain** [comment]
That is awesome. :+1: pending tests.

File: recipes/_ruby.rb [hunk 1]
```diff
@@ -61,11 +62,14 @@
     build_env['CFLAGS']  = '-O0 -g -pipe -mcpu=v9'
     build_env['LDFLAGS'] = '-mcpu=v9'
   end
+
+  patches << 'https://raw.githubusercontent.com/chef/omnibus-software/38e8befd5ecd14b7ad32c4bd3118fe4caf79ee92/config/patches/ruby/ruby-solaris-linux-socket-compat.patch'
```
Comment (curiositycasualty) [inline recipes/_ruby.rb]: Does it make sense to grab this file from the master branch?
If we're trying to lock the version of it, maybe use a git tag as the ref.

File: spec/recipes/ruby_spec.rb [hunk 2]
```diff
@@ -56,6 +56,11 @@
           )
       end
 
+      it 'applies a patch to the Ruby source' do
+        expect(chef_run).to install_ruby_install('2.1.5')
+          .with_patches(%w(https://raw.githubusercontent.com/chef/omnibus-software/38e8befd5ecd14b7ad32c4bd3118fe4caf79ee92/config/patches/ruby/ruby-solaris-linux-socket-compat.patch))
```
Comment (curiositycasualty) [inline spec/recipes/ruby_spec.rb]: Needs amendment if the above is altered.

**Comment by curiositycasualty** [comment]
:+1: Other than comments.

File: recipes/_ruby.rb (hunk 1 above)
Comment (schisamo) [inline recipes/_ruby.rb]: That does pull from master. It's just the canonical URL form of:
https://github.com/chef/omnibus-software/blob/master/config/software/ruby.rb

Comment (curiositycasualty) [inline recipes/_ruby.rb]: :+1:

**Comment by scotthain** [comment]
:shipit:

**Pull request merged**